arbitrum-docs/arbos/l1-to-l2-messaging.mdx

```diff
@@ -1,64 +1,159 @@
+    <Node id="1">Does the auto-redeem succeed?</Node>
+    <Node id="2">Ticket is executed</Node>
+    <Node id="3">Ticket is deleted</Node>
+    <Node id="4">excessFeeRefundAddress gets refunded</Node>
+    <Connection from="1" to="2" label="yes"/>
+    <Connection from="2" to="3"/>
+    <Connection from="1" to="4" label="no"/>
+  </Nodes>
+  <NodeDescriptions>
+    <NodeDescription for='1'><code>Does the auto-redeem succeed?</code> Logic that determines if the user's L2 Balance is greater than (or equal to) <code>l2Basefee * gasLimit</code> && <code>maxFeePerGas</code> is greater than (or equal to) the <code>l2Basefee</code></NodeDescription>
+    <NodeDescription for='2'><code>Ticket is executed</code> Ticket is executed, the actual  <code>submissionFee</code> is refunded to the  <code>excessFeeRefundAddress</code> since the ticket was not kept in the buffer on L2</NodeDescription>
+    <NodeDescription for='3'><code>Ticket is deleted</code> Ticket gets deleted from the L2 retryable buffer</NodeDescription>
+    <NodeDescription for='4'><code>excessFeeRefundAddress gets refunded</code> <code>excessFeeRefundAddress</code> gets refunded with <code>(maxGas - gasUsed) * gasPrice</code>. Note that this amount is capped by <code>gasUsed * baseFee</code> in the auto-reddem</NodeDescription>
+  </NodeDescriptions>
+</MermaidWithHtml>
+
+### Manual Redemption
+
+5. At this point, _anyone_ can attempt to manually redeem the ticket again by calling `ArbRetryableTx`'s redeem precompile method, which donates the call's gas to the next attempt. 
+ArbOS will [enqueue the redeem][enqueue_link], which is its own special `ArbitrumRetryTx` type, to its list of redeems that ArbOS [guarantees to exhaust][exhaust_link] before moving on to the next non-redeem transaction in the block its forming. In this manner redeems are scheduled to happen as soon as possible, and will always be in the same block as the tx that scheduled it. 
+Note that the redeem attempt's gas comes from the call to redeem, so there's no chance the block's gas limit is reached before execution.
+6. If the fixed period (one week) elapses without a successful redeem, the ticket **expires** and will be [automatically **discarded**][discard_link], unless some party has paid a fee to [**renew**][renew_link] the ticket for another full period. A ticket can live indefinitely as long as it is renewed each time before it expires.
 
+[enqueue_link]: https://github.com/OffchainLabs/nitro/blob/fa36a0f138b8a7e684194f9840315d80c390f324/arbos/block_processor.go#L245
+[exhaust_link]: https://github.com/OffchainLabs/nitro/blob/fa36a0f138b8a7e684194f9840315d80c390f324/arbos/block_processor.go#L135
 [discard_link]: https://github.com/OffchainLabs/nitro/blob/fa36a0f138b8a7e684194f9840315d80c390f324/arbos/retryables/retryable.go#L262
 [renew_link]: https://github.com/OffchainLabs/nitro/blob/fa36a0f138b8a7e684194f9840315d80c390f324/arbos/retryables/retryable.go#L207
```

Comment:
use keepalive instead of renew?
https://github.com/OffchainLabs/nitro-contracts/blob/a68783436b5105a64f54efe5fbd55174704a7618/src/precompiles/ArbRetryableTx.sol#L41